src/main/scala/io/citrine/lolo/trees/regression/RegressionTrainingLeaf.scala

```diff
@@ -0,0 +1,41 @@
+package io.citrine.lolo.trees.regression
+
+import io.citrine.lolo.trees.{ModelLeaf, ModelNode, TrainingNode}
+import io.citrine.lolo.{Learner, Model, PredictionResult}
+
+import scala.collection.mutable
+
+/**
+  * Created by maxhutch on 3/8/17.
+  */
+class RegressionTrainingLeaf(
+                              trainingData: Seq[(Vector[AnyVal], Double, Double)],
+                              leafLearner: Learner,
+                              depth: Int
+                            ) extends TrainingNode(trainingData, depth) {
+
+  /**
+    * Average the training data
+    *
+    * @return lightweight prediction node
+    */
+  def getNode(): ModelNode[PredictionResult[Double]] = {
+    new ModelLeaf(leafLearner.train(trainingData).getModel().asInstanceOf[Model[PredictionResult[Double]]], depth)
+  }
+
+  def getFeatureImportance(): scala.collection.mutable.ArraySeq[Double] = {
+    importance match {
+      case Some(x) =>
+        val expectations: (Double, Double) = trainingData.map{ case (v, l, w) =>
+          (l * w, l * l * w)
+        }.reduce((u: (Double, Double), v: (Double, Double)) => (u._1 + v._1, u._2 + v._2))
+        val impurity = expectations._2 / trainingData.size - Math.pow(expectations._1 / trainingData.size, 2.0)
```

Comment:
We don't actually calculate the impurity in the splitters.  This is a performance optimization: the splitter considers a very large number of splits, so the least it does for each one the better.

The importance isn't always used, so why compute it unless it is?  This is more of a memory optimization (feature importance can be larger than the split) than a run-time one.

We could expose other importance measures, but I'm not sure that using impurity is on a critical performance path (see memory comment above).  Although we don't compute the impurity for each split, we do compute the change in impurity, so that's already available.